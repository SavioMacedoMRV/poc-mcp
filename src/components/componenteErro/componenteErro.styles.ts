import styled from 'styled-components'
import {cores} from 'resources/cores'
import {margens} from 'resources/margens'
import {Texto} from 'components/texto'
import AlertaSVG from 'icons/iAlerta.svg'
import LupaSVG from 'icons/iLupa.svg'

export const Container = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const IconeAlerta = styled(AlertaSVG).attrs({
  height: 64,
  width: 64,
  fill: cores.neutralMedium,
})``

export const IconeLupa = styled(LupaSVG).attrs({
  height: 64,
  width: 64,
  fill: cores.neutralMedium,
})``

export const Mensagem = styled(Texto).attrs({
  tamanho: 16,
  alturalinha: 20,
  cursor: 'default',
  estilo: 'semibold',
  cor: cores.neutralXDark,
})`
  margin-top: ${margens.medium}px;
`
