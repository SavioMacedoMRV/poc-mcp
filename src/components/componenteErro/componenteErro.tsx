import React from 'react'
import * as S from './componenteErro.styles'

interface ComponenteErroProps {
  estadoVazio?: boolean
}
export function ComponenteErro({estadoVazio}: Readonly<ComponenteErroProps>) {
  return estadoVazio ? (
    <S.Container data-testid="mensagem-erro-estado-vazio">
      <S.IconeLupa />
      <S.Mensagem>Nenhum item foi encontrado</S.Mensagem>
    </S.Container>
  ) : (
    <S.Container data-testid="mensagem-erro">
      <S.IconeAlerta />
      <S.Mensagem>Houve um erro ao acessar o Portal 360</S.Mensagem>
    </S.Container>
  )
}
