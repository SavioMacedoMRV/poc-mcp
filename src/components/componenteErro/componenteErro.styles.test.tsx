import {render} from '@testing-library/react'
import * as S from './componenteErro.styles'

describe('ComponenteErro.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Container />)
    expect(container).toMatchSnapshot()
  })

  test('IconeAlerta DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeAlerta />)
    expect(container).toMatchSnapshot()
  })

  test('IconeLupa DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeLupa />)
    expect(container).toMatchSnapshot()
  })

  test('Mensagem DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Mensagem />)
    expect(container).toMatchSnapshot()
  })
})
