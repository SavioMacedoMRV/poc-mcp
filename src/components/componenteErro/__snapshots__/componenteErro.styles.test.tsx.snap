// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponenteErro.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`ComponenteErro.styles IconeAlerta DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgIAlerta"
    fill="#999E9C"
    height="64"
    width="64"
  />
</div>
`;

exports[`ComponenteErro.styles IconeLupa DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgILupa"
    fill="#999E9C"
    height="64"
    width="64"
  />
</div>
`;

exports[`ComponenteErro.styles Mensagem DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
  font-family: AvertaStd-Semibold;
  color: #434645;
  text-align: start;
  cursor: default;
  text-transform: none;
}

.c1 {
  margin-top: 24px;
}

<div>
  <span
    alturalinha="20"
    class="c0 c1"
    cor="#434645"
    cursor="default"
    estilo="semibold"
    tamanho="16"
  />
</div>
`;
